db/db.go

```diff
@@ -101,7 +101,15 @@ func (m *memKVStore) Delete(namespace string, key []byte) error {
 
 // Commit commits a batch
 func (m *memKVStore) Commit(b KVStoreBatch) (e error) {
+	succeed := false
 	b.Lock()
+	defer func() {
+		b.Unlock()
+		// clear the batch if commit succeeds
+		if succeed {
+			b.Clear()
+		}
+	}()
 	for i := 0; i < b.Size(); i++ {
 		write, err := b.Entry(i)
 		if err != nil {
```

Comment:
This is one bug for MemKVStore that the batch was not unlocked

Reply:
right, good catch!